### ISSUE TITLE
Fix S2589 FP: Change this condition so that it does not always evaluate to 'True'

### ISSUE THREAD

**Issue by paulhickman-ec** [SonarSource/sonar-dotnet#8449]
### Description

The code below erroneously reports 2589 Change this condition so that it does not always evaluate to 'True' on the expression ```delay < 409600```

### Repro steps

````C#
public static void Foo()
{
    int delay = 100;
    while (true)
    {
        if (delay < 409600) delay = delay * 2;
    }
}
````
### Expected behavior

No warning

### Actual behavior

Warning S2589

### Known workarounds

None

### Related information

* C#/VB.NET Plugins version 9.15.0.81779
* Visual Studio version 17.7
* MSBuild / dotnet version 7
* Operating System Windows 11

**Comment by mary-georgiou-sonarsource** [comment]
Hello @paulhickman-ec, 

Thanks for reporting this.
 I confirm this is an FP - I've added this issue to our backlog to our backlog.
